tests/test_plugins.py

```diff
@@ -0,0 +1,141 @@
+
+
+def _run_plugins(st,
+                 make_all=False,
+                 **proces_kwargs):
+    """
+    Try all plugins (except the DAQReader) for a given context (st) to see if
+    we can really push some (empty) data from it and don't have any nasty
+    problems like that we are referring to some non existant dali folder.
+    """
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        st.storage = [strax.DataDirectory(temp_dir)]
+
+        # As we use a temporary directory we should have a clean start
+        assert not st.is_stored(run_id, 'raw_records'), 'have RR???'
+
+        # Create event info
+        target = 'event_info'
+        st.make(run_id=run_id,
+                targets=target,
+                **proces_kwargs)
+
+        # The stuff should be there
+        assert st.is_stored(run_id, target), f'Could not make {target}'
+
+        # I'm only going to do this for nT because:
+        #  A) Doing this many more times does not give us much more
+        #     info (everything above already worked fine)
+        #  B) Most development will be on nT, 1T may get less changes
+        #     in the near future
+        if make_all:
+            # Now make sure we can get some data for all plugins
+            for p in list(st._plugin_class_registry.keys()):
+                if p not in forbidden_plugins:
+                    st.get_array(run_id=run_id,
+                                 targets=p,
+                                 **proces_kwargs)
+
+                    # Check for types that we want to save that they are stored.
+                    if (int(st._plugin_class_registry['peaks'].save_when) >
+                            int(strax.SaveWhen.TARGET)):
+                        is_stored = st.is_stored(run_id, p)
+                        assert is_stored, f"{p} did not save correctly!"
+
+    print("Wonderful all plugins work (= at least they don't fail), bye bye")
+
+
+def _update_context(st, max_workers):
+    # Change config to allow for testing both multiprocessing and lazy mode
+    st.set_context_config({'forbid_creation_of': forbidden_plugins})
+    st.register(DummyRawRecords)
+    if max_workers - 1:
+        st.set_context_config({
+            'allow_multiprocess': True,
+            'allow_lazy': False,
+            'timeout': 60,  # we don't want to build travis for ever
+        })
+
+##
+# Tests
+##
+
+
+def test_1T(ncores=1):
+    if ncores == 1:
+        print('-- 1T lazy mode --')
+    st = straxen.contexts.xenon1t_dali()
+    _update_context(st, ncores)
+    _run_plugins(st, make_all=False, max_wokers=ncores)
+    print(st.context_config)
+
+
+def test_nT(ncores=1):
+    if ncores == 1:
+        print('-- nT lazy mode --')
+    st = straxen.contexts.xenonnt_online(_database_init=False)
+    _update_context(st, ncores)
+    _run_plugins(st, make_all=True, max_wokers=ncores)
+    print(st.context_config)
+
+
+def test_nT_mutlticore():

```

Comment:
Do you see any reason to test things also in lazy mode? I do not, but on the other hand these tests should be super fast why not simply run them.

Reply:
I'd argue that we want to know if we only have issues in lazy/mutlicore mode as that would allow us to debug software faster. Also I don't see any particular valid reason not to test lazy mode